jsm.go

```diff
@@ -611,6 +627,23 @@ type StreamLister struct {
 	pageInfo *apiPaged
 }
 
+// Streams returns a receive only channel to iterate on the streams.
+func (js *js) StreamsInfo() <-chan *StreamInfo {
+	ach := make(chan *StreamInfo)
+	sl := &streamLister{js: js}
+	go func() {
```

Comment:
Feels like it should have timeouts and a way to cancel it etc no?

Reply:
sounds good, we could make it take options like, `nats.MaxWait()` and `nats.Context()` as I did with `AckSync`

Reply:
I tried adding options for both timeout and context (https://github.com/wallyqs/nats.go/commit/265ef17a0f0b00c5c7efce81cd34c0a95bfe2ff8) but now thinking that might be simpler if only support context.Context instead for this case, so change to something like:

```go
StreamsInfo(ctx context.Context) <-chan *StreamInfo
ConsumersInfo(ctx context.Context, stream string) <-chan *ConsumerInfo
```